neqo-transport/src/recovery.rs

```diff
@@ -464,8 +471,12 @@ impl LossRecovery {
         self.rtt_vals.pto()
     }
 
-    pub fn drop_0rtt(&mut self) -> impl Iterator<Item = SentPacket> {
-        self.spaces[PNSpace::ApplicationData].remove_ignored()
+    pub fn drop_0rtt(&mut self) -> Vec<SentPacket> {
+        let pkts: Vec<SentPacket> = self.spaces[PNSpace::ApplicationData]
```

Comment:
This demonstrates a problem with the signature of the `cc.remove_ignore()` function.  That should instead take `T: IntoIter<SentPacket>` so that you don't have to create a temporary `Vec` to hold the discarded packets.